AsyncDisplayKit/ASControlNode.mm

```diff
@@ -316,13 +298,22 @@ - (NSArray *)actionsForTarget:(id)target forControlEvent:(ASControlNodeEvent)con
 
   ASDN::MutexLocker l(_controlLock);
   
-  // Grab the event dispatch table for this event.
-  NSMapTable *eventDispatchTable = _controlEventDispatchTable[_ASControlNodeEventKeyForControlEvent(controlEvent)];
-  if (!eventDispatchTable)
+  // Grab the event target action array for this event.
+  NSMutableArray *eventTargetActionArray = _controlEventDispatchTable[_ASControlNodeEventKeyForControlEvent(controlEvent)];
+  if (!eventTargetActionArray) {
     return nil;
+  }
 
-  // Return the actions for this target.
-  return [eventDispatchTable objectForKey:target];
+  NSMutableArray *actions = [[NSMutableArray alloc] init];
+  
+  // Collect all actions for this target.
+  for (ASControlTargetAction *targetAction in eventTargetActionArray) {
+    if ((target == nil && targetAction.createdWithNoTarget) || (target != nil && target == targetAction.target)) {
+      [actions addObject:NSStringFromSelector(targetAction.action)];
```

Comment:
I think this would change the performance characteristics of this code quite a bit - won't this method get called on the main thread when event delivery is happening?  It would be better to build this upon adding the action, which may occur on a background thread, instead of always on the main thread.